wbooks/src/app/screens/bookDetail/index.js

```diff
@@ -0,0 +1,101 @@
+import React, {Component} from 'react';
+import booksJson from '../../../../resources/books.json';
+import {Link} from 'react-router-dom';
+import './style.css';
+import Comments from './components/Comments';
+import NotFoundPage from '../../components/NotFoundPage';
+
+class BookDetail extends Component {
+
+  constructor({match}){
+    super();
+    this.state = {book: this.getBookInfo(match.params.id),
+      comments: []};
+  }
```

Comment:
remove the costructor override, and define the state as `state = { comments: [], book: null };`

and then write the method `componentWillMount` to get the book info and set the state

Reply:
check this page to learn about component lifecycle ;) 

https://reactjs.org/docs/react-component.html